### ISSUE TITLE
macOS Catalina: Stuck on 'Installing vanilla launcher' and unable to start Mojang Launcher

### ISSUE THREAD

**Issue by dafri92** [CreeperHost/modpacklauncher#14]
I am using App Version 202006271817-a3b3b73d4cd46fc91f731c0a18f918d9eadda1a8-release on macOS Catalina 10.15.6.

Trying to install a mod pack (with freshly installed FTB App) gets stuck on step 'Installing vanilla launcher' ([Log file](https://pste.ch/loqukokido)).
After digging in the code and finding this line https://github.com/CreeperHost/modpacklauncher/blob/a3b3b73d4cd46fc91f731c0a18f918d9eadda1a8/src/main/java/net/creeperhost/creeperlauncher/minecraft/GameLauncher.java#L56 I checked my data folder under `~./ftba/bin` and no file `launcher_profiles.json` was created. So I copied over this JSON file from my vanilla Minecraft installation.

After this the installation of the mod pack continued and finished ([Log file](https://pste.ch/benawoyaja)).
So I tried starting the freshly installed mod pack which had no visible effect and didn't start the Minecraft Launcher ([Log file](https://pste.ch/sefimukile)).

So I tried starting it myself manually from the shell by running in my data directory `./Minecraft.app/Contents/MacOS/launcher --workDir /Users/daniel/.ftba/bin`. This started the Minecraft Launcher and the mod pack profile was visible and I was able to play.

I am suspecting that the installation of the Minecraft Launcher got stuck, because the Minecraft Launcher was never started and therefore `launcher_profiles.json` was never created.

TL;DR
For some reason on macOS starting the Minecraft Launcher from the FTB App fails.

**Comment by Frostist** [comment]
I have the same issue. Doesn't ever load past Installing Vanilla Launcher, and sometimes the loading bar for vanilla launcher doubles, (See Screenshot)
\Except when I launch Minecraft from .ftba/bin it works.

![Screenshot 2020-07-17 at 10 29 00 AM](https://user-images.githubusercontent.com/37545822/87765635-55a70180-c818-11ea-9491-a6ca6d7245bb.png)

**Comment by ThePaul-T** [comment]
We have a proc that we call that launches it in the background then closes it again to generate the json.

I think this may be something related to some of the filesystem issues we've been finding recently (Java seems to dislike working across multiple file systems, and as the ftbapp 'image' is technically it's own filesystem, I'm gonna guess it's complaining because of it, alternatively, it's just the launch code not working on mac, but it was tested, but has been changed since then to allow for alternate storage locations)

I'll get someone on it ASAP, it'll likely be solved for the next preview release which we're hoping to have available this coming Wednesday, if it isn't, it'll be the next Wednesday after.

**Comment by Frostist** [comment]
Thank you so much, It has been such a pain, I hope it gets fixed soon!😋

**Comment by Cloudhunter** [comment]
Can you try this build? It won't work, but it has extra logging (hopefully)

Please note that the launcher_profiles.json will need to not exist in order to get any useful logging as to what might be going wrong.

https://apps.modpacks.ch/FTBApp/develop/202007171236-c79927a016-develop/FTBA_macos_202007171236-c79927a016-develop.dmg

**Comment by dafri92** [comment]
First I deleted .ftba and settings by running
```
rm -r .ftba
rm -r Library/Application\ Support/ftb-app
```

then I started your development build and tried to install a mod pack. Unfortunately I got stuck at "Creating environment..." ([Log](https://pste.ch/ifuparapav)).

Here is a screenshot of the GUI:
![Screenshot 2020-07-17 at 16 54 35](https://user-images.githubusercontent.com/930308/87800368-9cafe980-c84e-11ea-9ef8-fb160c304942.png)

Did I forget to delete some previously created files?

**Edit**
After closing the electron GUI, the downloaded Minecraft Launch dmg appeared in `.ftba/bin` and was mounted ([Log](https://pste.ch/musetaxidi)).

**Comment by Cloudhunter** [comment]
Somehow it is crashing earlier in the process than even doing anything with the Vanilla launcher, and providing no useful logging.

I've added more logging and catching of some exceptions in this build. Can you try it out for me?

https://apps.modpacks.ch/FTBApp/debug-200/202007171659-61bb7d85f0-debug-200/FTBA_macos_202007171659-61bb7d85f0-debug-200.dmg

**Comment by dafri92** [comment]
Sure. Deleted settings again and tried to install a mod pack ([Log](https://pste.ch/sebujomome)). Again stuck on "Creating environment...".

![Screenshot 2020-07-17 at 18 26 49](https://user-images.githubusercontent.com/930308/87809301-76dd1180-c85b-11ea-8d2b-af1f4614df25.png)

A folder with random uuid is created in .`ftba/instances` and it contains the art.png and instance.json.
The only file `.ftba/bin` is settings.json.

**Comment by dafri92** [comment]
I started the FTB App manually by running `java -jar launcher-202007171659-61bb7d85f0-debug-200-all.jar` in `/Applications/FTBApp.app/Contents/Resources/app`. I tried to install a mod pack again and got stuck. After closing the app the following log appeared in my shell as well as in launcher.log file.
```
Jul 17, 2020 6:47:20 PM net.creeperhost.creeperlauncher.CreeperLogger error
SEVERE: Not a valid instance '.localCache', skipping...
Jul 17, 2020 6:47:20 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: Loaded 0 out of 1 instances.
Jul 17, 2020 6:49:49 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: closed /127.0.0.1:51928 with exit code 1001 additional info: 
Jul 17, 2020 6:49:49 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: Starting stage INIT
Jul 17, 2020 6:49:49 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: FTB Interactions 5 87
Jul 17, 2020 6:49:49 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: Starting stage VANILLA
Jul 17, 2020 6:49:49 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: Starting download of the vanilla launcher
Jul 17, 2020 6:49:49 PM net.creeperhost.creeperlauncher.CreeperLogger warning
WARNING: Vanilla already exists.
Jul 17, 2020 6:49:51 PM net.creeperhost.creeperlauncher.CreeperLogger info
INFO: Launching Vanilla launcher and closing - path and args: /Users/daniel/.ftba/bin/Minecraft.app/Contents/MacOS/launcher --workDir /Users/daniel/.ftba/bin
```

**Comment by vhd** [comment]
Hi there. Same issues under macOS
Just tried debug version of a launcher (https://github.com/CreeperHost/modpacklauncher/issues/14#issuecomment-660188970)
([Log](https://pste.ch/vezabidote))
It seems stuck on trying to run vanilla launcher.
But then i've tried to run launcher manually
```
vhd@wsh09729 .ftba % ./bin/Minecraft.app/Contents/MacOS/launcher --workDir /Users/vhd/.ftba/bin 
2020-07-21 20:40:09.797 launcher[6276:465951] Claimed it found a path: /Users/vhd/.ftba/./bin/Minecraft.app/Contents/MacOS/launcher (/Users/vhd/.ftba/bin/Minecraft.app/Contents/MacOS/launcher) 4096
https://launchermeta.mojang.com/v1/products/launcher/022631aeac4a9addbce8e0503dce662152dc198d/mac-os.json
2020-07-21 20:40:10.193 nativeUpdater[6277:466012] Found executable path: /Users/vhd/.ftba/bin/Minecraft.app/Contents/MacOS/launcher
2020-07-21 20:40:10.207 nativeUpdater[6277:466012] Found executable path: /Users/vhd/.ftba/bin/Minecraft.app/Contents/MacOS/launcher
https://launchermeta.mojang.com/v1/products/launcher/022631aeac4a9addbce8e0503dce662152dc198d/mac-os.json
OK
OK
```
And now it kinda works?.. Except i'm unable to launch any of installed packs.
<img width="1435" alt="Снимок экрана 2020-07-21 в 20 42 37" src="https://user-images.githubusercontent.com/487056/88088941-e3297f00-cb93-11ea-8e48-7c89f2269441.png">

**Comment by Frostist** [comment]
I have copied the vanilla launcher files from my normal Minecraft to the folder in Library, but that doesn't even work. It gets past installing but then doesn't launch at all.

**Comment by JippeJoosten** [comment]
I got the error that `launcher_profiles.json` did not exist in the bin file, after adding it and a restart it started installing! After that, I had to open Minecraft client once, reinstall the mod pack and it started to work. I still can't launch from the client but I can using the command `...@....-MBP .ftba % ./bin/Minecraft.app/Contents/MacOS/launcher --workDir /Users/.../.ftba/bin `

**Comment by dafri92** [comment]
Hi,
I just tried with a fresh install of the current preview (202012101145-0e92584e6c-preview) and installation of modpacks on macOS Catalina (10.15.7) works now. Thank you!
